salesagility/SuiteCRM
# Fixed #8443 - Improve LanguageManager::clearLanguageCache()

## Description




Issue reference: #8443

Checks if an entry is a subdirectory of 'cache/modules/' before calling LanguageManager::_clearCache().

## Motivation and Context

The function LanguageManager::clearLanguageCache() causes PHP-FPM to throw a warning
when PHP-FPM has an enabled 'open_basedir' restriction and the directory 'cache/modules/' contains files.
This occurs because above function attempts to remove e.g. 'cache/modules/unified_search_modules.php/language/en_us.lang.php' through the subfunction LanguageManager::_clearCache().

## How To Test This

1. Set up PHP-FPM with 'open_basedir' restriction:
E.g. in '/etc/php/7.1/fpm/pool.d/www.conf' put
`php_admin_value[open_basedir] = /var/www/html/suitecrm`
2. Restart PHP-FPM: e.g. by executing `service php7.1-fpm restart`
3. Do a "Quick Repair and Rebuild" in SuiteCRM
4. Check Apache error log for messages like
```
PHP Warning:  file_exists(): open_basedir restriction in effect. File(cache/modules/unified_search_modules.php/language/en_us.lang.php)
is not within the allowed path(s): (/var/www/html/suitecrm) in /var/www/html/suitecrm/include/SugarObjects/LanguageManager.php on line 179
```

## Types of changes


### Final checklist
